node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/maintenance/acl/FilterTableLineEditor.java

```diff
@@ -29,15 +29,33 @@ static FilterTableLineEditor from(Acl acl, IPVersion ipVersion) {
 
     @Override
     public LineEdit edit(String line) {
-        // We have already added all the lines we wanted, remove the remainer
-        if (wantedRules.isEmpty()) return LineEdit.remove();
-
-        String wantedRule = wantedRules.pop();
-        return wantedRule.equals(line) ? LineEdit.none() : LineEdit.replaceWith(wantedRule);
+        int index = indexOf(wantedRules, line, position);
+        // Unwanted rule, remove
+        if (index < 0) return LineEdit.remove();
+
+        // Wanted rule at the expected position, no diff
+        if (index == position) {
+            position++;
+            return LineEdit.none();
+        }
+
+        // Insert the rules between position and index before index
+        List<String> toInsert = wantedRules.subList(position, index);
+        position = ++index;
+        return LineEdit.insertBefore(toInsert);
     }
 
     @Override
     public List<String> onComplete() {
-        return this.wantedRules;
+        return position == wantedRules.size() ? List.of() : wantedRules.subList(position, wantedRules.size());
```

Comment:
You can always return `wantedRules.subList(position, wantedRules.size())`, even if `position == wantedRules.size()`.